### PR TITLE
Change RHCOS os_version schema to match rhcos4, rhcos4.X and rhcos4.X.Y

### PR DESCRIPTION
**Pull request by gyohuangxin**

### CODE REVIEW

**Comment by jichenjc** [comment]
please add additional test cases .. such as 4.6.12, 4.12.14, 4.6.8 , 4.6 , 4 pass test and 4.6.a,  4.a fail test

File: zvmsdk/sdkwsgi/validation/parameter_types.py [hunk 1]
```diff
@@ -390,7 +391,7 @@ def valid_char(char):
  '^((u|U)(b|B)(u|U)(n|N)(t|T)(u|U))(16|20){1}([.][0-9]{2})?([.][0-9])?$'},
  {'type': 'string',
  'pattern':
- '^((r|R)(h|H)(c|C)(o|O)(s|S))(4){1}([.][0-9])?$'}
+ '^((r|R)(h|H)(c|C)(o|O)(s|S))(4){1}([.][0-9])?([.][0-9])?$'}
```
Comment (bjhuangr) [inline zvmsdk/sdkwsgi/validation/parameter_types.py:394]: does this change support input os_version as `4.11.11` ?

Comment (gyohuangxin) [reply zvmsdk/sdkwsgi/validation/parameter_types.py:394]: No, it only support one number. I will change it to support double numbers.

Comment (jichenjc) [reply zvmsdk/sdkwsgi/validation/parameter_types.py:394]: as I commented above, please make sure tests such as 
`please add additional test cases .. such as 4.6.12, 4.12.14, 4.6.8 , 4.6 , 4 pass test and 4.6.a, 4.a fail test` 
and you may consider more UT test cases

Comment (gyohuangxin) [reply zvmsdk/sdkwsgi/validation/parameter_types.py:394]: @jichenjc Yes, thanks.  I have tested these cases and it works as expected.

Pass:
```
786dc385-4652-488e-b11a-30e880104b21 rhcos4.12.14 ACTIVE None Running flat01=172.26.104.13 rhcos4.12.14 b95ffbd8-5bf1-4974-bd6c-e81399472126   Default Group BOEM5404 os_distro='RHCOS4.12.14'
f6198818-6691-4fff-8cfc-bc891cbae055 rhcos4.7.11 ACTIVE None Running flat01=172.26.104.38 rhcos4.7.11 9666532b-b405-49c3-ab00-99ae1e521bde   Default Group BOEM5404 os_distro='RHCOS4.7.11'
17bc8602-7b59-411c-9dee-d9391df538e5 rhcos4 ACTIVE None Running flat01=172.26.104.37 rhcos4 5f1a6ce4-9102-42dd-a0ea-3c909c9dce18   Default Group BOEM5404 os_distro='RHCOS4'
a3c87783-02c7-48c6-9e9d-8f32972bfce1 rhcos4.7.0 ACTIVE None Running flat01=172.26.104.78 rhcos4.7.0 c3566a9b-3a48-4aae-90bf-abd5ef65c7e3   Default Group BOEM5404 os_distro='RHCOS4.7.0'
```

Fail:
```
780c4c8f-c41e-4b61-8dc4-149a6255f4e4 rhcos4.678.678 ERROR None NOSTATE  rhcos4.678.678 2c460c79-5681-4bbd-a787-2930a4649d85    None os_distro='RHCOS4.678.678'
343b1d66-a416-4cef-97f8-01474f224eb3 rhcos4.a ERROR None NOSTATE   d6696af6-11ed-431e-9905-11413bbeb3ec    None os_distro='RHCOS4.a'
49ddb39b-b092-46a5-82c8-981ecf2d7571 rhcos4.6.a ERROR None NOSTATE   2100eba3-275e-4100-93fa-886eafdf722a    None os_distro='RHCOS4.6.a'
a2bed965-c3db-499f-b8e2-5c87c8fc7f22 rhcos4. ERROR None NOSTATE   0b4ee3a2-ab7d-4f88-9dd1-270a7bb69f38    None os_distro='RHCOS4.'
```

And for UT cases, do you mean that I need add above test cases in `zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py`?

**Comment by gyohuangxin** [comment]
@jichenjc @bjhuangr I've added above UT test cases to check the validation of rhcos os version. Pls review again.

File: zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py [hunk 2]
```diff
@@ -850,6 +850,250 @@ def test_guest_delete_network_interface(self, mock_interface, mock_userid):
...
+        self.assertRaises(exception.ValidationError,
+                       guest.guest_create_network_interface,
+                       self.req)
+
+    @mock.patch.object(util, 'wsgi_path_item')
+    @mock.patch('zvmconnector.connector.ZVMConnector.send_request')
+    def test_guest_create_network_interface_with_wrong_rhcos_version_3(self,
+                                                       mock_interface,
+                                                       mock_userid):
+        bstr = """{"interface": {"os_version": "rhcos4.111.111",
```
Comment (jichenjc) [inline zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py:1081]: I think it's good enough

except this 4.111.111 is valid ? not a big deal at all

Comment (gyohuangxin) [reply zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py:1081]: 4.xxx.xxx isn't valid. Double digits are supported at most.

**Comment by jichenjc** [comment]
thanks~

**Pull request merged**